subprojects/wrapper/src/main/java/org/gradle/wrapper/Download.java

```diff
@@ -66,23 +65,28 @@ private void downloadInternal(URI address, File destination)
             in = conn.getInputStream();
             byte[] buffer = new byte[BUFFER_SIZE];
             int numRead;
-            int contentLength = conn.getContentLength();
-            long progressCounter = 0;
-            long numDownloaded = 0;
+            long totalLength = conn.getContentLengthLong();
+            long downloadedLength = 0;
             while ((numRead = in.read(buffer)) != -1) {
                 if (Thread.currentThread().isInterrupted()) {
                     System.out.print("interrupted");
                     throw new IOException("Download was interrupted.");
                 }
-                numDownloaded += numRead;
-                progressCounter += numRead;
-                if (progressCounter / PROGRESS_CHUNK > 0) {
-                    logger.append(".");
-                    progressCounter = progressCounter - PROGRESS_CHUNK;
-                    if (progressListener != null) {
-                        progressListener.downloadStatusChanged(address, contentLength, numDownloaded);
-                    }
+
+                long previousDownloadedLength = downloadedLength;
+                downloadedLength += numRead;
+
+                int currentDownloadPercentTenth = calculateDownloadPercent(totalLength, downloadedLength) / 10;
+                int previousDownloadPercentTenth = calculateDownloadPercent(totalLength, previousDownloadedLength) / 10;
```

Comment:
instead of calculating it on every iteration you can store previous value outside of loop right :)